### ISSUE TITLE
Error Launching Holochain

### ISSUE THREAD

**Issue by bhaugen** [holochain/launcher-tauri#55]
Error Launching Holochain
This holochain version is not running: LaunchHolochainError(CouldNotConnectToConductor("Connection refused (os error 111)"))

**To Reproduce**
From https://github.com/lightningrodlabs/rea-playspace
1. Download and install the latest [Holochain Launcher](https://github.com/holochain/launcher/releases/tag/v0.4.7).
2. Start Launcher
3. Give it a password
4. Restart launcher with new password
5. result: errot message above

**Expected behavior**

 Follow the prompts until you get to the main screen with the button 'INSTALL NEW APP' then 'SELECT APP FROM FILESTORE'. Select the rea-playspace.webhapp file from the previous step.

Use the default AppId. Select 0.0.136 for the Holochain version to install.

Advanced: If you want to leave the UID option empty, you can. This will connect you to a network with everyone else who has also left the option empty. If you want to create a playspace with only a few others, enter a UID (any string of characters) and share it with them.

Click 'Install App'. Find the app in the Installed Apps view. Click open. This will launch it in the browser.

**Logs**
Please attach here the `launcher.log` file. To find it, go to `Holochain Admin` window of the Holochain Launcher, select the  `Settings` menu, and select `Open Logs`. Attach that file to this issue.

> In Ubuntu, there is a weird error that prevents the upload of a file from inside a hidden folder. To work around it, you can copy the `launcher.log` file to another non-hidden location, and upload it from there. 
{See launcher log below)

**Screenshots**

**Desktop (please complete the following information):**
 - OS: Ubuntu 18.04.6 LTS
 - Holochain Launcher Version  v0.4.7
 - 
[launcher.log](https://github.com/holochain/launcher/files/8950600/launcher.log)

**Comment by guillemcordoba** [comment]
Thanks for the report @bhaugen .

Does the issue still persist even after a computer reboot?

**Comment by bhaugen** [comment]
> Does the issue still persist even after a computer reboot?

Nope. Should have tried that immediately. Sorry to waste your time.

**Comment by guillemcordoba** [comment]
No no you didn't waste my time, this is still an issue...

This is related to children holochain process still being alive after the launcher has been closed. The Launcher tries to launch Holochain in the same port and crashes, because the old process is still running in that port. I thought I handled this correctly up until now, but it seems there is still some edge case in which this doesn't work. 

I think I can implement better protection against this, thanks :)

**Comment by matthme** [comment]
The same error can also occur if the conductor is not ready in time when the `HolochainManager` tries to connect to the admin websocket. We should try to add a handshake there. For now I added a workaround to retry connecting with the admin websocket one more time (866adff2655d772c0ed3e1ab9a8390ae9c7f893f)
